fe/fe-core/src/main/java/com/starrocks/http/HttpServerHandler.java

```diff
@@ -103,7 +104,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 if (LOG.isDebugEnabled()) {
                     LOG.debug("action: {} ", action.getClass().getName());
                 }
-                if (action.supportAsyncHandler()) {
+                // Health Action need to be handled synchronously
+                if (action.supportAsyncHandler() && !(action instanceof HealthAction)) {
```

Comment:
shall override the `supportAsyncHandler()` in HealthAction(), return false, so it won't go with the async code path.

Reply:
You are right. I missed this function.